File: frontend/src/types/risk.ts
import type { PageQuery } from './common'

export interface RiskRecord {
  id: number
  fingerprint?: string
  risk_type: string
  risk_level: string
  title?: string
  project_id?: number
  project_name?: string
  task_id?: number
  task_name?: string
  user_id?: number
  user_name?: string
  status: string
  detail?: string
  source_data?: Record<string, unknown>
  detected_at: string
  due_at?: string
  handled_by?: number
  handler_name?: string
  handled_at?: string
  handling_note?: string
  resolved_at?: string
  created_at: string
  updated_at: string
}

export interface RiskQuery extends PageQuery {
  status?: string
  risk_type?: string
  risk_level?: string
  project_id?: number
  user_id?: number
}

export interface RiskStats {
  total: number
  open: number
  resolved: number
  by_status: Record<string, number>
  by_level: Record<string, number>
}

File: backend/app/schemas/risk.py
from datetime import datetime

from pydantic import Field

from app.schemas.common import ORMModel


class RiskHandleRequest(ORMModel):
    status: str
    handling_note: str = Field(min_length=1, max_length=2000)


class RiskResponse(ORMModel):
    id: int
    fingerprint: str | None
    risk_type: str
    risk_level: str
    title: str | None
    project_id: int | None
    project_name: str | None = None
    task_id: int | None
    task_name: str | None = None
    user_id: int | None
    user_name: str | None = None
    status: str
    detail: str | None
    source_data: dict | None
    detected_at: datetime
    due_at: datetime | None
    handled_by: int | None
    handler_name: str | None = None
    handled_at: datetime | None
    handling_note: str | None
    resolved_at: datetime | None
    created_at: datetime
    updated_at: datetime

